File: frontend/src/app/model/application/comment/comment-type.ts
export enum CommentType {
  INVOICING = 'INVOICING',
  RETURN = 'RETURN',
  REJECT = 'REJECT',
  INTERNAL = 'INTERNAL',
  PROPOSE_APPROVAL = 'PROPOSE_APPROVAL',
  PROPOSE_REJECT = 'PROPOSE_REJECT',
  PROPOSE_TERMINATION = 'PROPOSE_TERMINATION',
  TO_EXTERNAL_SYSTEM = 'TO_EXTERNAL_SYSTEM'
}

export const manualComments: Array<CommentType> = [
  CommentType.INVOICING,
  CommentType.INTERNAL,
  CommentType.TO_EXTERNAL_SYSTEM
];

export const decisionProposalComments: Array<CommentType> = [
  CommentType.PROPOSE_APPROVAL,
  CommentType.PROPOSE_REJECT,
  CommentType.PROPOSE_TERMINATION
];

File: backend/common/src/main/java/fi/hel/allu/common/types/CommentType.java
package fi.hel.allu.common.types;

/**
 * Valid comment types for application comments
 */
public enum CommentType {
  INVOICING(true),
  RETURN(false),
  REJECT(true),
  INTERNAL(true),
  PROPOSE_APPROVAL(false),
  PROPOSE_REJECT(false),
  PROPOSE_TERMINATION(false),
  EXTERNAL_SYSTEM(false),
  TO_EXTERNAL_SYSTEM(true);

  private final boolean manuallyAdded;

  private CommentType(boolean manuallyAdded) {
    this.manuallyAdded = manuallyAdded;
  }

  public boolean isManuallyAdded() {
    return manuallyAdded;
  }
}
